### PR TITLE
Added retry logic while Deleting SearchParameter

### PR DESCRIPTION
**Pull request by apurvabhaleMS**
## Description
Adding a retry logic while Deleting SearchParameter.
When there are multiple instances of the fhir-server running, it could take some time for the search parameter/reindex updates to propogate to all instances. Hence we are adding some retries below to account for that delay.

## Related issues
Addresses [[82891](https://microsofthealth.visualstudio.com/Health/_workitems/edit/82891)].

## Testing
Manual Testing
Unit Test

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [x] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [x] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core/Features/Search/SearchParameterNotSupportedException.cs [hunk 1]
```diff
@@ -45,6 +45,7 @@ public SearchParameterNotSupportedException(string resourceType, string paramNam
         /// Initializes a new instance of the <see cref="SearchParameterNotSupportedException"/> class.
         /// </summary>
         /// <param name="definitionUri">The search parameter definition URL.</param>
+        /// <param name="location">The search parameter location of the error.</param>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.Health.Fhir.Core/Features/Search/SearchParameterNotSupportedException.cs:48]: >/// <param name="location">The search parameter location of the error.</param> [](http://example.com/codeflow?start=8&length=78)

this seems off

File: src/Microsoft.Health.Fhir.Tests.Common/TestFiles/Normative/Appointment.json [hunk 2]
```diff
@@ -47,7 +47,6 @@
             }
         ]
     },
-    "priority": 5,
```
Comment (Ivanidzo4ka) [inline src/Microsoft.Health.Fhir.Tests.Common/TestFiles/Normative/Appointment.json:50]: >"priority": 5, [](http://example.com/codeflow?start=4&length=14)

as alternative you can add new file to R5 folder with priority been CodeableConcept


"priority":
 {
    "coding": [
      {
        "system": "[http://terminology.hl7.org/ValueSet/v3-ActPriority](http://terminology.hl7.org/ValueSet/v3-ActPriority)",
        "code": "EM",
        "display": "emergency"
      }
    ],
    "text": "Emergency"
  }

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs [hunk 3]
```diff
@@ -486,7 +485,29 @@ private async Task DeleteSearchParameterAndVerify(SearchParameter searchParam)
...
+                {
+                    success = true;
+                    retryCount++;
+                    try
+                    {
+                        await Client.DeleteAsync(searchParam);
+                    }
+                    catch (Exception exp)
+                    {
+                        _output.WriteLine($"Failed to delete searchParameter: {exp}");
```
Comment (Ivanidzo4ka) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs:504]: >exp [](http://example.com/codeflow?start=79&length=3)

Would it contain search parameter url or any other identifier?

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs [hunk 4]
```diff
@@ -186,6 +184,7 @@ public async Task GivenASearchParam_WhenUpdatingParam_ThenResourcesIndexedWithUp
                 Assert.Equal(randomName, param.Value.ToString());
 
                 // When job complete, search for resources using new parameter
+                await Task.Delay(10000);
```
Comment (Ivanidzo4ka) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs:187]: >await Task.Delay(10000); [](http://example.com/codeflow?start=16&length=24)

why we wait here and not after we post search parameter?

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs (hunk 3 above)
Comment (apurvabhaleMS) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs:504]: yes it does

File: src/Microsoft.Health.Fhir.Core/Features/Search/SearchParameterNotSupportedException.cs (hunk 1 above)
Comment (apurvabhaleMS) [reply src/Microsoft.Health.Fhir.Core/Features/Search/SearchParameterNotSupportedException.cs]: yes missed reverting this back. should be good now

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs (hunk 4 above)
Comment (apurvabhaleMS) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs]: added the wait after job is posted

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs [hunk 5]
```diff
@@ -175,6 +174,8 @@ public async Task GivenASearchParam_WhenUpdatingParam_ThenResourcesIndexedWithUp
 
                 // Reindex just a single patient, so we can try searching with a partially indexed search param
                 (reindexJobResult, reindexJobUri) = await Client.PostReindexJobAsync(new Parameters(), $"Patient/{expectedPatient.Resource.Id}/");
+                await Task.Delay(10000);
```
Comment (rbans96) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs:177]: nit: await Task.Delay(TimeSpan.FromSeconds(10)); for better readability

**Review by Ivanidzo4ka — APPROVED** [review]
:shipit:

**Pull request merged**